workspace_integration_test.go

```diff
@@ -2644,3 +2648,47 @@ func TestWorkspace_DataRetentionPolicy(t *testing.T) {
 		require.Nil(t, dataRetentionPolicy)
 	})
 }
+
+func TestWorkspacesAutoDestroy(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	t.Cleanup(orgTestCleanup)
+
+	upgradeOrganizationSubscription(t, client, orgTest)
+
+	autoDestroyAt := Time(time.Date(2025, 1, 1, 0, 0, 0, 0, time.UTC))
+	wTest, wCleanup := createWorkspaceWithOptions(t, client, orgTest, WorkspaceCreateOptions{
+		Name:          String(randomString(t)),
+		AutoDestroyAt: autoDestroyAt,
+	})
+	t.Cleanup(wCleanup)
+
+	assert.Equal(t, wTest.AutoDestroyAt, autoDestroyAt)
+
+	// respect default omitempty
+	w, err := client.Workspaces.Update(ctx, orgTest.Name, wTest.Name, WorkspaceUpdateOptions{
+		AutoDestroyAt: nil,
+	})
+
+	require.NoError(t, err)
+	assert.NotNil(t, w.AutoDestroyAt)
+
+	// explicitly update the value of auto_destroy_at
+	w, err = client.Workspaces.Update(ctx, orgTest.Name, wTest.Name, WorkspaceUpdateOptions{
+		AutoDestroyAt: NullableTime(time.Date(2025, 1, 2, 0, 0, 0, 0, time.UTC)),
+	})
```

Comment:
💅 Generally, I like to use `require` over `assert` when it doesn't make sense to continue the test if the assertion fails. `assert` will allow test execution to continue.